File: src/main/java/com/patres/timetable/repository/SubjectRepository.java
package com.patres.timetable.repository;

import com.patres.timetable.domain.Subject;
import com.patres.timetable.domain.Teacher;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Repository;

import org.springframework.data.jpa.repository.*;

import java.util.List;

@SuppressWarnings("unused")
@Repository
public interface SubjectRepository extends DivisionOwnerRepository<Subject> {

}

File: src/main/kotlin/com/patres/timetable/repository/SubjectRepository.kt
package com.patres.timetable.repository

import com.patres.timetable.domain.Subject
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import org.springframework.stereotype.Repository

@Repository
interface SubjectRepository : DivisionOwnerRepository<Subject> {

    @Query("""
        select
            subject
        from
            Subject subject
            left join fetch subject.preferenceSubjectByPlace
            left join fetch subject.preferredDivisions
            left join fetch subject.preferenceSubjectByTeacher
            left join fetch subject.preferencesDataTimeForSubject
        where
            subject.id =:id
        """)
    fun findOneWithPreference(@Param("id") id: Long?): Subject?

}
